.github/actions/create-pr/create_pr_for_staged_changes.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+set -uo pipefail # prevent accessing unset env vars, prevent masking pipeline errors to the next command
+
+#docs
+#title              :create_pr_for_staged_changes.sh
+#description        :This script will create a PR for staged changes, detect and close duplicate PRs. All PRs will be omitted from Release Notes and Changelogs
+#author		    :@heitorlessa
+#date               :May 8th 2023
+#version            :0.1
+#usage		    :bash create_pr_for_staged_changes.sh {git_staged_files_or_directories_separated_by_space}
+#notes              :Meant to use in GitHub Actions only. Temporary branch will be named $TEMP_BRANCH_PREFIX-$GITHUB_RUN_ID
+#os_version         :Ubuntu 22.04.2 LTS
+#required_env_vars  :PR_TITLE, TEMP_BRANCH_PREFIX, GH_TOKEN
+#==============================================================================
+
+# Sets GitHub Action with error message to ease troubleshooting
+function error() {
+    echo "::error file=${FILENAME}::$1"
+    exit 1
+}
+
+function debug() {
+    TIMESTAMP=$(date -u "+%FT%TZ") # 2023-05-10T07:53:59Z
+    echo ""${TIMESTAMP}" - $1"
+}
+
+function notice() {
+    echo "::notice file=${FILENAME}::$1"
+}
+
+function start_span() {
+    echo "::group::$1"
+}
+
+function end_span() {
+    echo "::endgroup::"
+}
+
+function has_required_config() {
+    start_span "Validating required config"
+    test -z "${TEMP_BRANCH_PREFIX}" && error "TEMP_BRANCH_PREFIX env must be set to create a PR"
+    test -z "${PR_TITLE}" && error "PR_TITLE env must be set"
+    test -z "${GH_TOKEN}" && error "GH_TOKEN env must be set for GitHub CLI"
+
+    # Default GitHub Actions Env Vars: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+    debug "Are we running in GitHub Action environment?"
+    test -z "${GITHUB_RUN_ID}" && error "GITHUB_RUN_ID env must be set to trace Workflow Run ID back to PR"
+    test -z "${GITHUB_SERVER_URL}" && error "GITHUB_SERVER_URL env must be set to trace Workflow Run ID back to PR"
+    test -z "${GITHUB_REPOSITORY}" && error "GITHUB_REPOSITORY env must be set to trace Workflow Run ID back to PR"
+
+    debug "Config validated successfully!"
+    set_environment_variables
+    end_span
+}
+
+function set_environment_variables() {
+    start_span "Setting environment variables"
+    export readonly WORKFLOW_URL="${GITHUB_SERVER_URL}"/"${GITHUB_REPOSITORY}"/actions/runs/"${GITHUB_RUN_ID}" # e.g., heitorlessa/aws-lambda-powertools-test/actions/runs/4913570678
```

Comment:
Should we change the example repo in this comment to something more generic rather than a an existing GitHub repo tied to a real user?

Reply:
Good point, changed it